tests/hazmat/primitives/test_pkcs12.py

```diff
@@ -282,6 +292,42 @@ def _load_ca(backend):
     reason="PKCS12 unsupported in FIPS mode. So much bad crypto in it."
 )
 class TestPKCS12Creation:
+    @pytest.mark.parametrize(
+        ("ktype", "kparam"),
+        [
+            pytest.param(
+                ed448.Ed448PrivateKey,
+                None,
+                marks=pytest.mark.supported(
+                    only_if=lambda backend: backend.ed448_supported(),
+                    skip_message="Requires OpenSSL with Ed448 support",
+                ),
+            ),
+            pytest.param(
+                ed25519.Ed25519PrivateKey,
+                None,
+                marks=pytest.mark.supported(
+                    only_if=lambda backend: backend.ed25519_supported(),
+                    skip_message="Requires OpenSSL with Ed25519 support",
+                ),
+            ),
+            (rsa.RSAPrivateKey, (65537, 1024)),
+            # (rsa.RSAPrivateKey, (65537, 2048)),
+            # (rsa.RSAPrivateKey, (65537, 3072)),
+            # (rsa.RSAPrivateKey, (65537, 4096)),
+            (dsa.DSAPrivateKey, (1024,)),
+            # (dsa.DSAPrivateKey, (2048,)),
+            # (dsa.DSAPrivateKey, (3072,)),
+            # (dsa.DSAPrivateKey, (4096,)),
+            (ec.EllipticCurvePrivateKey, (ec.SECP256R1,)),
```

Comment:
I don't think we really need every possible type here, but there is a curve supported check: https://github.com/pyca/cryptography/blob/main/src/cryptography/hazmat/backends/openssl/backend.py#L1183

Reply:
my instinct was to test all curves which the docs don't recommend against, but ill follow your lead.  which should be tested?

Reply:
If you want to add them all I have no objection, you just need to do a curve supported check.